sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -66,6 +66,9 @@ private[sql] object H2Dialect extends JdbcDialect {
     case _ => JdbcUtils.getCommonJDBCType(dt)
   }
 
+  override def convertBigDecimalToDecimal(d: BigDecimal, precision: Int, scale: Int): Decimal =
+    Decimal(d)
```

Comment:
This seems wrong. I think we should make sure the final `Decimal` instance we return has the same precision and scale as the JDBC column type.